syncer/syncer.go

```diff
@@ -1895,6 +1952,9 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 
 			job := newXIDJob(currentLocation, startLocation, currentLocation)
 			err2 = s.addJobFunc(job)
+			if waitXIDStatus(s.waitXIDJob.Load()) == waitComplete {
```

Comment:
is this branch needed 🤔

Reply:
If remove this branch, when dose `runCtx.cancel()`?

Reply:
there's another check under `s.getEvent`, at line 1740